### PR TITLE
expose signature to hex via rosetta.ts

### PR DESCRIPTION
**Pull request by robrobbins**
This is a draft of the possible solution for the issue/RFC #1404 

in draft as tests are still being written

### CODE REVIEW

File: src/mina-signer/src/rosetta.ts [hunk 1]
```diff
@@ -20,6 +20,14 @@ function signatureFromHex(signatureHex: string): Signature {
   };
 }
```
Comment (robrobbins) [inline src/mina-signer/src/rosetta.ts:22]: my particular use case is that i will have a payload from `client.signTransaction`. this it's expected that a user would call `Signature.fromJSON(signature)` first and pass that to the method below

File: src/mina-signer/src/rosetta.ts [hunk 2]
```diff
@@ -20,6 +20,14 @@ function signatureFromHex(signatureHex: string): Signature {
...
 
+function signatureToHex(signature: Signature): string {
+  // TODO possibly check isOdd like public key to hex does?
```
Comment (mitschabaude) [inline src/mina-signer/src/rosetta.ts:24]: not necessary here - "isOdd" is part of a compressed public key, but not part of a signature

**Review by mitschabaude — COMMENTED** [review]
I noticed that `fieldToHex` seems to have two major bugs (my bad):
- It uses AND instead of OR when putting in the padding bit
- It doesn't pad a zero when turning a byte to hex

Can you fix this while you're here?

```diff
function fieldToHex<T extends Field | Scalar>(
  binable: Binable<T>,
  x: T,
  paddingBit: boolean = false
) {
  let bytes = binable.toBytes(x);
  // set highest bit (which is empty)
-  bytes[bytes.length - 1] &= Number(paddingBit) << 7;
+  bytes[bytes.length - 1] |= Number(paddingBit) << 7;
  // map each byte to a hex string of length 2
  return bytes
-    .map((byte) => byte.toString(16).split('').reverse().join(''))
+    .map((byte) => byte.toString(16).padStart(2, '0').split('').reverse().join(''))
    .join('');
}
```

**Comment by robrobbins** [comment]
going to close this and re-open with a fresh branch after #1414 is merged

**Pull request closed without merging**